optimum/graphcore/ipu_configuration.py

```diff
@@ -34,6 +35,13 @@
 ALLOWED_POD_TYPES = ["pod4", "pod8", "pod16", "pod32", "pod64"]
 
 
+class IPUConfigAttributeTypes:
+    T = TypeVar("T")
+    Type: TypeAlias = Union[T, Dict[str, T]]
+    OptionalType = Union[Optional[T], Dict[str, Optional[T]]]
```

Comment:
Maybe this? Or do we need to specify that some `T`s in a dict can be None too?
```suggestion
    OptionalType = Union[T, Dict[str, T], None]
```

Reply:
Its possible that they are None. e.g. `inference_ipus_per_replica = None` by default